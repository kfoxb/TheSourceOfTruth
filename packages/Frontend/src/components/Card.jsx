import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Divider, Header, Icon, Image, Segment } from 'semantic-ui-react';

const StyledSegment = styled(Segment)`
  @media (max-width: 600px) {
    height: 150px;
  }

  @media (min-width: 600px) {
    height: 400px;
    width: 300px;
  }
`;

const StyledBody = styled.div`
  @media (max-width: 600px) {
    margin-left: 142px;
    margin-right: 17px;
    position: absolute;
    top: 40px;
    width: 500px;
  }
`;

const StyledDiv = styled.div`
  bottom: 0px;
  opacity: 0.5;
  position: absolute;

  @media (max-width: 600px) {
    margin-left: 142px;

  }

  @media (min-width: 600px) {
    width: 90%;
  }
`;

const StyledImage = styled(Image)`
  object-fit: cover;

  @media (max-width: 600px) {
    height: 150px;
    left: -14px;
    position: absolute;
    top: -14px;
    width: 137px;
  }
`;

export default function Card(props) {
  return (
    <StyledSegment>
      <Icon
        name="star"
        onClick={props.toggleFav}
        size="huge"
        style={{
          color: props.starFilled ? 'yellow' : 'whitesmoke',
          position: 'absolute',
          zIndex: 1,
          right: 0,
          top: 9,
        }}
      />
      <StyledImage />
      <StyledBody>
        <Header>
            props.title
        </Header>
        <p>
          This is the description
        </p>
      </StyledBody>
      <StyledDiv>
        <Divider />
        <p style={{ display: 'inline-block' }} >
          <Icon name="file text outline" />
          Read Time
        </p>
        <p style={{ display: 'inline-block', float: 'right' }}>
          Published 06/16/18
        </p>
      </StyledDiv>
    </StyledSegment>
  );
}

Card.propTypes = {
  toggleFav: PropTypes.func.isRequired,
  starFilled: PropTypes.bool.isRequired,
};
