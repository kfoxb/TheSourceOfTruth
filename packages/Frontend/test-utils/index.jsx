/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    const wrapper = shallow(<Component {...props} />);
    expect(wrapper).toMatchSnapshot();
  };
