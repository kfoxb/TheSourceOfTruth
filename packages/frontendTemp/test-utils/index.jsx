/* eslint-env jest */
/* eslint-disable import/no-extraneous-dependencies */
import React from 'react';
import { shallow } from 'enzyme';

export const makeShallowCreateWrapper = (Component, defaultProps) =>
  (extraProps) => {
    const props = { ...defaultProps, ...extraProps };
    return shallow(<Component {...props} />);
  };

export const createAssertWithPropsToMatchSnapshot = (Component, defaultProps) => {
  const createWrapper = makeShallowCreateWrapper(Component, defaultProps);
  return (extraProps) => {
    expect(createWrapper(extraProps)).toMatchSnapshot();
  };
};
