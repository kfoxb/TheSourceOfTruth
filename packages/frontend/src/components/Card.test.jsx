/* eslint-env jest */
import { createAssertWithPropsToMatchSnapshot } from '../../test-utils';
import Card from './Card';

describe('Card', () => {
  const defaultProps = {
    starFilled: false,
    toggleFav: () => {},
  };
  const assertWithPropsToMatchSnapshot =
    createAssertWithPropsToMatchSnapshot(Card, defaultProps);
  it('should render', () => {
    assertWithPropsToMatchSnapshot();
  });

  it('should render as star filled when passed starFilled as a prop', () => {
    assertWithPropsToMatchSnapshot({ starFilled: true });
  });
});
