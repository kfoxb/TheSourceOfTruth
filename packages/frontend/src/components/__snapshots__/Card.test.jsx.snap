// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render 1`] = `
<Styled(Segment)>
  <Icon
    as="i"
    name="star"
    onClick={[Function]}
    size="huge"
    style={
      Object {
        "color": "whitesmoke",
        "position": "absolute",
        "right": 0,
        "top": 9,
        "zIndex": 1,
      }
    }
  />
  <Styled(Image) />
  <styled.div>
    <Header>
      props.title
    </Header>
    <p>
      This is the description
    </p>
  </styled.div>
  <styled.div>
    <Divider />
    <p
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <Icon
        as="i"
        name="file text outline"
      />
      Read Time
    </p>
    <p
      style={
        Object {
          "display": "inline-block",
          "float": "right",
        }
      }
    >
      Published 06/16/18
    </p>
  </styled.div>
</Styled(Segment)>
`;

exports[`Card should render as star filled when passed starFilled as a prop 1`] = `
<Styled(Segment)>
  <Icon
    as="i"
    name="star"
    onClick={[Function]}
    size="huge"
    style={
      Object {
        "color": "yellow",
        "position": "absolute",
        "right": 0,
        "top": 9,
        "zIndex": 1,
      }
    }
  />
  <Styled(Image) />
  <styled.div>
    <Header>
      props.title
    </Header>
    <p>
      This is the description
    </p>
  </styled.div>
  <styled.div>
    <Divider />
    <p
      style={
        Object {
          "display": "inline-block",
        }
      }
    >
      <Icon
        as="i"
        name="file text outline"
      />
      Read Time
    </p>
    <p
      style={
        Object {
          "display": "inline-block",
          "float": "right",
        }
      }
    >
      Published 06/16/18
    </p>
  </styled.div>
</Styled(Segment)>
`;
